gammapy/irf/effective_area_table.py

```diff
@@ -0,0 +1,572 @@
+            raise ValueError("Energies must be Quantity objects.")
+        if not isinstance(offset_lo, Angle) or not isinstance(offset_hi, Angle):
+            raise ValueError("Offsets must be Angle objects.")
+        if not isinstance(eff_area, Quantity) or not isinstance(eff_area_reco, Quantity):
+            raise ValueError("Effective areas must be Quantity objects.")
+
+        self.energ_lo = energ_lo.to('TeV')
+        self.energ_hi = energ_hi.to('TeV')
+        self.offset_lo = offset_lo.to('degree')
+        self.offset_hi = offset_hi.to('degree')
+        self.eff_area = eff_area.to('m^2')
+        self.eff_area_reco = eff_area_reco.to('m^2')
+
+        # actually offset_hi = offset_lo
+        self.offset = (offset_hi + offset_lo) / 2
+        self.energy = np.sqrt(energ_lo * energ_hi)
+
+        self._prepare_linear_interpolator()
+        self._prepare_spline_interpolator()
+
+        # set to linear interpolation by default
+        self.interpolation_method = 'linear'
+
+    @staticmethod
+    def from_fits(hdu_list):
+        """Create OffsetDependentEffectiveAreaTable from ``GCTAAeff2D`` format HDU list.
+
+        Parameters
+        ----------
+        hdu_list : `~astropy.io.fits.HDUList`
+            HDU list with ``EFFECTIVE AREA`` extension.
+
+        Returns
+        -------
+        eff_area : `OffsetDependentEffectiveAreaTable`
+            Offset dependent Effective Area object.
+        """
+        e_lo = Quantity(
+            hdu_list['EFFECTIVE AREA'].data['ENERG_LO'][0, :], 'TeV')
+        e_hi = Quantity(
+            hdu_list['EFFECTIVE AREA'].data['ENERG_HI'][0, :], 'TeV')
+        o_lo = Angle(
+            hdu_list['EFFECTIVE AREA'].data['THETA_LO'][0, :], 'degree')
+        o_hi = Angle(
+            hdu_list['EFFECTIVE AREA'].data['THETA_HI'][0, :], 'degree')
+        ef = Quantity(
+            hdu_list['EFFECTIVE AREA'].data['EFFAREA'][0, :, :], 'm^2')
+        efrec = Quantity(
+            hdu_list['EFFECTIVE AREA'].data['EFFAREA_RECO'][0, :, :], 'm^2')
+
+        return OffsetDependentEffectiveAreaTable(e_lo, e_hi, o_lo, o_hi, ef, efrec)
+
+    @staticmethod
+    def read(filename):
+        """Read FITS format effective area file (``GCTAAeff2D`` output).
+
+        Parameters
+        ----------
+        filename : str
+            File name
+
+        Returns
+        -------
+        eff_area : `OffsetDependentEffectiveAreaTable`
+            Offset dependent Effective Area object.
+
+        """
+        hdu_list = fits.open(filename)
+        return OffsetDependentEffectiveAreaTable.from_fits(hdu_list)
+
+    def evaluate(self, offset=None, energy=None):
+        """Evalute effective area for a given energy and offset
+
+        If a certain parameter is not given, the nodes from the FITS table are used.
+        2D input arrays are not supported yet.
+
+        Parameters
+        ----------
+        offset : Angle
```

Comment:
Use this:
```
offset : `~astropy.coordinates.Angle`
```
Then if you do `python setup.py build_sphinx` the data type will be a link to the corresponding class in the Astropy docs, which I think is convenient and warrants the extra typing.